kernel/src/main/java/org/kframework/utils/InterrupterRunnable.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2019 K Team. All Rights Reserved.
+package org.kframework.utils;
+
+/**
+ * A runnable that interrupts the given thread when invoked, then awaits termination for the given wait time, in ms.
+ *
+ * @author Denis Bogdanas
+ * Created on 24-Dec-18.
+ */
+public class InterrupterRunnable implements Runnable {
+
+    private final Thread thread;
+    private final int waitTime;
+
+    public InterrupterRunnable(Thread thread, int waitTime) {
+        this.thread = thread;
+        this.waitTime = waitTime;
+    }
+
+    @Override
+    public void run() {
+        thread.interrupt();
+        try {
+            thread.join(waitTime);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
This is not an acceptable catch block. You should either raise a KEMException or else call Thread.currentThread.interrupt() if you want to swallow the exception. In general we should never just call printStackTrace in a catch block, that's a sign you have written boilerplate code without properly considering what should happen if the exception occurs.

Reply:
The exception happens in a shutdown hook. If I interrupt the current thread, this will be ignored. I suspect throwing another exception in this thread also will be ignored. Notifying the user with a printStackTrace, particularly in this context, I believe is the best option.

Reply:
Alright, that makes sense.